### ISSUE TITLE
Generic.WhiteSpace.ScopeIndent false positive when using ternary operator with short arrays

### ISSUE THREAD

**Issue by SpazzMarticus** [squizlabs/PHP_CodeSniffer#2702]
`Generic.WhiteSpace.ScopeIndent` expects 4 more whitespace characters when using a ternary operator:

```php
function failure($a)
{
    $a = $a === true ? [
        'a' => 1,
        'b' => 2,
        ] : [
        'a' => 100,
        'b' => 99,
    ]; //Line indented incorrectly; expected at least 8 spaces, found 4
}
```

Alternatives using `?:` or `??` work fine with only 4 whitespace characters:

```php
function success1($a)
{
    $a = $a === true ?: [
        'a' => 1,
        'b' => 2,
    ];
}

function success2($a)
{
    $a = $a === true ?? [
        'a' => 1,
        'b' => 2,
    ];
}
``` 

 Is this expected behaviour?

**Comment by gsherwood** [comment]
Thanks for reporting this. There was a problem specifically checking short arrays in this case. The sniff was trying to align the closing bracket with the line that the opening bracket is on (hence the 8 spaces required) instead of with the start of the statement (`$a = ...`), which only required 4 spaces.

Fix will be in 3.5.3
